environment-dev.yml

```diff
@@ -1,36 +1,23 @@
-name: testenv
+name: pymc3-dev
 channels:
   - conda-forge
   - defaults
 dependencies:
-  - python>=3.6
+  - python=3.6
```

Comment:
Why did you fix this?

Reply:
I was getting conflicts when I removed that, made an environment with Python 3.7, and then updated from the file, see #4281

I _think_ we should maybe make three different environment files: `environment-dev-py36.yml`, `environment-dev-py37.yml`, and `environment-dev-py38.yml`

Reply:
just tried that, and it works fine (no conflicts!), so that's what I've pushed

Reply:
I don't think that's necessary. Can't we pin 38 here instead?

Reply:
As in, pin 3.8 here and then run the tests using that?

_could_ do, but I would've thought it would be better to run the tests on the earliest supported version of Python, else someone could accidentally use something modern (e.g. the walrus operator), tests pass, but then pymc3 breaks when people install it using Python3.6.

Like this, the full test suite can be run with `environment-dev-py36.yml`, the arviz compatitiblity tests with `environment-dev-py38.yml`, and hopefully most bases will be covered

Reply:
Oh, so `dev` is used to specify the GH action env?

I was thinking this is for people who want to set up a dev environment locally. I suppose it can be both in which case your suggestion of having different envs for different python versions would make sense.

Reply:
yup, that's right - check `.github/workflows/pytest.yml`

```
      - uses: conda-incubator/setup-miniconda@v2
        with:
          activate-environment: pymc3-dev-py36
          channel-priority: strict
          environment-file: conda-envs/environment-dev-py36.yml
          use-only-tar-bz2: true # IMPORTANT: This needs to be set for caching to work properly!
```